src/development/add-to-app/ios/add-flutter-screen.md

```diff
@@ -44,7 +44,43 @@ the app delegate.
 
 {% samplecode engine %}
 
-{% sample Swift %}
+{% sample SwiftUI %}
+As an example, we demonstrate creating a
+`FlutterEngine` inside an `ObservableObject`, 
+and passing that into a `View` using the 
+ `environmentObject()` modifier. 
+
+ **In `MyApp.swift`:**
+ <!--code-excerpt "MyApp.swift" title-->
+ ```swift
+import SwiftUI
+import Flutter
+// Used to connect plugins (only if you have plugins with iOS platform code).
+import FlutterPluginRegistrant
+
+class FlutterDependencies: ObservableObject {
+  let npsFlutterEngine = FlutterEngine(name: "my flutter engine")
```

Comment:
nit: maybe just `flutterEngine`? readers may not have the context of newsfeed app (iirc this nps is from the news feed demo app?)

Reply:
Oops yeah my bad, thanks!